### DISCUSSION TITLE
Persistent database on local filesystem

### DISCUSSION THREAD

**Discussion by glacial-ai** [duckdb/duckdb-wasm#1387] [Q&A]
Hi! I know persistent DBs weren't supported when the WASM API was first released (see note in <https://duckdb.org/2021/10/29/duckdb-wasm.html>), but was wondering if there have been any updates since then to allow persistence?

**Comment by joshuatreepc** [comment]
@glacial-ai  Why do you need a persisted database on a local filesystem? What you need is persistent access to files... so persist the filehandle using index-db and then use partial filereads from DuckDB to read the files at will.

**Comment by yoonghm** [comment]
Does it mean that a JS script ran from node.js would have its "saved" data silently remove when the database is closed?
